Packs/Carbon_Black_Enterprise_Response/Integrations/CarbonBlackResponseV2/CarbonBlackResponseV2.py

```diff
@@ -728,49 +740,60 @@ def endpoint_command(client: Client, id: str = None, ip: str = None, hostname: s
 
 
 def fetch_incidents(client: Client, max_results: int, last_run: dict, first_fetch_time: str, status: str = None,
-                    feedname: str = None, query: str = None):
+                    feedname: str = None, query: str = ''):
     if (status or feedname) and query:
         raise Exception(f'{INTEGRATION_NAME} - Search is not permitted with both query and filter parameters.')
 
     max_results = arg_to_number(arg=max_results, arg_name='max_fetch', required=False) if max_results else 50
 
     # How much time before the first fetch to retrieve incidents
     first_fetch_time = dateparser.parse(first_fetch_time)
-    first_fetch_timestamp_ms = int(first_fetch_time.timestamp()) if first_fetch_time else None
-    last_fetch = last_run.get('last_fetch', None)
+    last_fetch = last_run.get('last_fetch', None)  # {last_fetch: timestamp}
+    demisto.debug(f'{INTEGRATION_NAME} - last fetch: {last_fetch}')
+
     # Handle first fetch time
     if last_fetch is None:
-        last_fetch = first_fetch_timestamp_ms
+        last_fetch = first_fetch_time
     else:
-        last_fetch = int(last_fetch)
+        last_fetch = datetime.fromtimestamp(last_fetch)
+
+    latest_created_time = last_fetch.timestamp()
+
+    date_range = f'[{last_fetch.strftime("%Y-%m-%dT%H:%M:%S")} TO *]'
 
-    latest_created_time = last_fetch
-    demisto.debug(f'{INTEGRATION_NAME} - last fetch: {last_fetch}')
     incidents: List[Dict[str, Any]] = []
 
+    alerts = []
+
     # multiple statuses are not supported by api. If multiple statuses provided, gets the incidents for each status.
     # Otherwise will run without status.
-    alerts = []
+    query_params = {'created_time': date_range}
+    if feedname:
+        query_params['feedname'] = feedname
+
     if status:
         for current_status in argToList(status):
             demisto.debug(f'{INTEGRATION_NAME} - Fetching incident from Server with status: {current_status}')
-            res = client.get_alerts(status=current_status, feedname=feedname)
+            query_params['status'] = current_status
+            # we create a new query containing params since we do not allow both query and params.
+            res = client.get_alerts(query=_create_query_string(query_params), limit=max_results)
```

Comment:
Why add it under the `query_params` instead of using the function args `feedname` and `status`?
Should they be removed from the function signature?

Reply:
as I stated in the comment (if it is not clear I would change it)- we create a new query containing params since we do not allow both query and params.
The get_alerts function is being used for other command as well, where it can get the feedname or status as parameters, or a query.
Since we do not allow both, It will fail if we provide in the fetch both query and other params, and since we must provides dates in the query and it is not a user-created query, we create a new query with both dates and the user params.